docs_test.go

```diff
@@ -0,0 +1,82 @@
+package cli
+
+import (
+	"testing"
+)
+
+func testApp() *App {
+	app := NewApp()
+	app.Name = "greet"
+	app.Flags = []Flag{
+		StringFlag{
+			Name:  "socket, s",
+			Usage: "some usage text",
+			Value: "value",
+		},
+		StringFlag{Name: "flag, fl, f"},
+		BoolFlag{
+			Name:  "another-flag, b",
+			Usage: "another usage text",
+		},
+	}
+	app.Commands = []Command{{
+		Aliases: []string{"c"},
+		Flags: []Flag{
+			StringFlag{Name: "flag, fl, f"},
+			BoolFlag{
+				Name:  "another-flag, b",
+				Usage: "another usage text",
+			},
+		},
+		Name:  "config",
+		Usage: "another usage test",
+		Subcommands: []Command{{
+			Aliases: []string{"s", "ss"},
+			Flags: []Flag{
+				StringFlag{Name: "sub-flag, sub-fl, s"},
+				BoolFlag{
+					Name:  "sub-command-flag, s",
+					Usage: "some usage text",
+				},
+			},
+			Name:  "sub-config",
+			Usage: "another usage test",
+		}},
+	}, {
+		Aliases: []string{"i", "in"},
+		Name:    "info",
+		Usage:   "retrieve generic information",
+	}, {
+		Name: "some-command",
+	}}
+	app.UsageText = "app [first_arg] [second_arg]"
+	app.Usage = "Some app"
+	app.Author = "Harrison"
+	app.Email = "harrison@lolwut.com"
+	app.Authors = []Author{{Name: "Oliver Allen", Email: "oliver@toyshop.com"}}
+	return app
+}
+
+func TestToMarkdown(t *testing.T) {
+	// Given
+	app := testApp()
+
+	// When
+	_, err := app.ToMarkdown()
+
+	// Then
+	// TODO: extend test case
+	expect(t, err, nil)
+}
+
+func TestToMan(t *testing.T) {
+	// Given
+	app := testApp()
+
+	// When
+	_, err := app.ToMan()
+
+	// Then
+	// TODO: extend test case
+	expect(t, err, nil)
```

Comment:
You could check in expected output, or dump it in here. Atleast the md stuff would be viewable in github.

Reply:
Yes, I add this as well.